spec/rspec_matchers/schema_spec.rb

```diff
@@ -0,0 +1,123 @@
+require 'rails_helper'
+
+module ActiveModelSerializers
+  module Test
+    module SchemaTest
+      class MyController < ActionController::Base
+        def index
+          render json: profile
+        end
+
+        def show
+          index
+        end
+
+        def name_as_a_integer
+          profile.name = 1
+          index
+        end
+
+        def render_using_json_api
+          render json: profile, adapter: :json_api
+        end
+
+        def invalid_json_body
+          render json: ''
+        end
+
+        private
+
+        def profile
+          @profile ||= Profile.new(name: 'Name 1', description: 'Description 1', comments: 'Comments 1')
+        end
+      end
+    end
+  end
+end
+
+RSpec.describe ActiveModelSerializers::Test::SchemaTest::MyController, type: :controller do
```

Comment:
needs request spec. see https://github.com/rails-api/active_model_serializers/pull/1947/files#r96015833

Also, the test app should be defined inside the RSpec block or namespace

this globally creates a `ActiveModelSerializers::Test::SchemaTest::MyController`

Might be a good case to start a dummy app like  in https://github.com/rails-api/active_model_serializers/pull/1414